### PR TITLE
Pv2 3233 record funding platform for payments generated via submit learner data platform

### PR DESCRIPTION
**Pull request by abroodie**

### CODE REVIEW

File: src/SFA.DAS.Payments.Application/Data/Configurations/PaymentModelConfiguration.cs [hunk 1]
```diff
@@ -60,6 +60,7 @@ public void Configure(EntityTypeBuilder<PaymentModel> builder)
             builder.Property(x => x.ReportingAimFundingLineType).HasColumnName(@"ReportingAimFundingLineType").IsRequired();
             builder.Property(x => x.LearningAimSequenceNumber).HasColumnName(@"LearningAimSequenceNumber");
             builder.Property(x => x.AgeAtStartOfLearning).HasColumnName(@"AgeAtStartOfLearning");
+            builder.Property(x => x.FundingPlatformType).HasColumnName("FundingPlatformType");
```
Comment (LiamWright) [inline src/SFA.DAS.Payments.Application/Data/Configurations/PaymentModelConfiguration.cs:63]: Does it need to be "@FundingPlatformType"

File: src/SFA.DAS.Payments.FundingSource.Application.UnitTests/Mapper/CalculatedRequiredLevyAmountMapperTest.cs [hunk 2]
```diff
@@ -9,9 +9,109 @@
...
+                OnProgrammeEarningType = OnProgrammeEarningType.Learning,
+                LearningAim = new LearningAim
+                {
+                    FrameworkCode = 403
+                },
+                PriceEpisodeIdentifier = "1819-P01",
+                SfaContributionPercentage = 0.9m,
+                Ukprn = 10000,
+
+                //AgreementId = "11",
```
Comment (LiamWright) [inline src/SFA.DAS.Payments.FundingSource.Application.UnitTests/Mapper/CalculatedRequiredLevyAmountMapperTest.cs:49]: Left in comments, but I'm not overly precious about that

File: src/SFA.DAS.Payments.FundingSource.Application/Infrastructure/Configuration/AutoMapperConfigurationFactory.cs [hunk 3]
```diff
@@ -60,16 +65,22 @@ public static MapperConfiguration CreateMappingConfig()
                 cfg.CreateMap<CalculatedRequiredOnProgrammeAmount, EmployerCoInvestedFundingSourcePaymentEvent>()
                     .Include<CalculatedRequiredCoInvestedAmount, EmployerCoInvestedFundingSourcePaymentEvent>()
                     .ForMember(dest => dest.FundingSourceType, opt => opt.UseValue(FundingSourceType.CoInvestedEmployer))
-                    .ForMember(dest => dest.AgeAtStartOfLearning, opt => opt.MapFrom(source => source.AgeAtStartOfLearning)); ;
+                    .ForMember(dest => dest.AgeAtStartOfLearning, opt => opt.MapFrom(source => source.AgeAtStartOfLearning))
+                    //.ForMember(dest => dest.FundingPlatformType, opt => opt.Ignore())
```
Comment (LiamWright) [inline src/SFA.DAS.Payments.FundingSource.Application/Infrastructure/Configuration/AutoMapperConfigurationFactory.cs:69]: Comments to remove

File: src/SFA.DAS.Payments.FundingSource.Messages/Commands/CalculateOnProgrammePayment.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        public long? ApprenticeshipId { get; set; }
+        public long? ApprenticeshipPriceEpisodeId { get; set; }
+        public ApprenticeshipEmployerType ApprenticeshipEmployerType { get; set; }
+        public DateTimeOffset EventTime { get; set; }
+        public Guid EventId { get; set; }
+        public Learner Learner { get; set; }
+        public LearningAim LearningAim { get; set; }
+        public CollectionPeriod CollectionPeriod { get; set; }
+        public FundingPlatformType FundingPlatformType { get; set; }
+        public int AgeAtStartOfLearning { get; set; }
```
Comment (LiamWright) [inline src/SFA.DAS.Payments.FundingSource.Messages/Commands/CalculateOnProgrammePayment.cs:37]: This might be nullable? Need to check

Comment (David-B-Read) [reply src/SFA.DAS.Payments.FundingSource.Messages/Commands/CalculateOnProgrammePayment.cs:37]: It's not nullable in the message issued from Payment Simplification (or any other future source), they need to supply the Funding Platform Type

File: src/SFA.DAS.Payments.Application/Data/Configurations/PaymentModelConfiguration.cs (hunk 1 above)
Comment (David-B-Read) [reply src/SFA.DAS.Payments.Application/Data/Configurations/PaymentModelConfiguration.cs]: It seems to work without the '@' but I've updated so it's consistent

File: src/SFA.DAS.Payments.FundingSource.Application.UnitTests/Mapper/CalculatedRequiredLevyAmountMapperTest.cs (hunk 2 above)
Comment (David-B-Read) [reply src/SFA.DAS.Payments.FundingSource.Application.UnitTests/Mapper/CalculatedRequiredLevyAmountMapperTest.cs]: Comments removed

File: src/SFA.DAS.Payments.FundingSource.Application/Infrastructure/Configuration/AutoMapperConfigurationFactory.cs (hunk 3 above)
Comment (David-B-Read) [reply src/SFA.DAS.Payments.FundingSource.Application/Infrastructure/Configuration/AutoMapperConfigurationFactory.cs]: Removed commented out mappings

**Review by LiamWright — DISMISSED** [review]

**Pull request merged**